### ISSUE TITLE
[feature request] Callback on learning rate drop in torch.optim.lr_scheduler.ReduceLROnPlateau

### ISSUE THREAD

**Issue by reynoldscem** [pytorch/pytorch#5737]
Currently [torch.optim.lr_scheduler.ReduceLROnPlateau](http://pytorch.org/docs/master/optim.html#torch.optim.lr_scheduler.ReduceLROnPlateau) has the option to print a fixed message once the learning rate drops.

I propose that either this be replaced, or modified in some way so that we can run an arbitrary closure / callback at each drop.

This could be useful for things such as logging which epochs the drop happened at, for terminating after a given number of drops, for taking a model snapshot, etc.

**Comment by StephanHeijl** [comment]
This behaviour can be emulated by checking if the `ReduceLROnPlateau` property `cooldown_counter` is equal to `ReduceLROnPlateau.cooldown` immediately after performing a step, or doing a simple boolean check on the `in_cooldown` property like so:

```
scheduler = optim.lr_scheduler.ReduceLROnPlateau(*args)
if scheduler.in_cooldown:
    callback()
```

**Comment by 0phoff** [comment]
+1  
While @StephanHeijl's method works, having a custom callback would be a cleaner interface imo.

**Comment by YashasSamaga** [comment]
any news about this?

**Comment by StephanHeijl** [comment]
I'd be willing to take a look at this, will give it a shot around the end of the week.
